libbeat/processors/processor.go

```diff
@@ -51,38 +51,54 @@ func New(config PluginConfig) (*Processors, error) {
 }
 
 func (procs *Processors) add(p Processor) {
-	procs.list = append(procs.list, p)
+	procs.List = append(procs.List, p)
 }
 
-// Applies a sequence of processing rules and returns the filtered event
-func (procs *Processors) Run(event common.MapStr) common.MapStr {
+// RunBC (run backwards-compatible) applies the processors, by providing the
+// old interface based on common.MapStr.
+// The event us temporarily converted to beat.Event. By this 'conversion' the
+// '@timestamp' field can not be accessed by processors.
+// Note: this method will be removed, when the publisher pipeline BC-API is to
+//       be removed.
+func (procs *Processors) RunBC(event common.MapStr) common.MapStr {
+	ret := procs.Run(&beat.Event{Fields: event})
+	if ret == nil {
+		return nil
+	}
+	return ret.Fields
+}
 
+// Applies a sequence of processing rules and returns the filtered event
+func (procs *Processors) Run(event *beat.Event) *beat.Event {
 	// Check if processors are set, just return event if not
-	if len(procs.list) == 0 {
+	if len(procs.List) == 0 {
 		return event
 	}
 
-	// clone the event at first, before starting filtering
-	filtered := event.Clone()
-	var err error
-
-	for _, p := range procs.list {
-		filtered, err = p.Run(filtered)
+	for _, p := range procs.List {
+		var err error
+		event, err = p.Run(event)
 		if err != nil {
+			// XXX: We don't drop the event, but continue filtering here iff the most
```

Comment:
Same here. Old/Current logic does ignore errors and at most prints a debug message. I don't intend to change any processing logic.